src/main/java/gregtech/common/blocks/BlockMetalCasing.java

```diff
@@ -1,53 +1,83 @@
 package gregtech.common.blocks;
 
+import codechicken.lib.render.CCRenderState;
+import codechicken.lib.render.pipeline.IVertexOperation;
+import codechicken.lib.texture.TextureUtils.IIconRegister;
+import codechicken.lib.vec.Cuboid6;
+import codechicken.lib.vec.Matrix4;
+import gregtech.api.render.ICubeRenderer;
+import gregtech.api.render.Textures;
+import gregtech.api.unification.material.MaterialIconType;
+import gregtech.api.unification.material.type.SolidMaterial;
+import net.minecraft.block.Block;
 import net.minecraft.block.SoundType;
 import net.minecraft.block.material.Material;
 import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.client.renderer.texture.TextureMap;
 import net.minecraft.entity.EntityLiving.SpawnPlacementType;
-import net.minecraft.util.IStringSerializable;
+import net.minecraft.util.EnumFacing;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.IBlockAccess;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
 
-public class BlockMetalCasing extends VariantBlock<BlockMetalCasing.MetalCasingType> {
+public final class BlockMetalCasing extends Block  implements ICubeRenderer, IIconRegister {
```

Comment:
It is straight wrong. Why do you need to implement ICubeRenderer from scratch? Why do you implement it within block glass?

Reply:
Yes, you right, I speed my day on that. The multiblock has a method called `public ICubeRenderer getBaseTexture(IMultiblockPart sourcePart)` And I'm not able to understand  how I can get an `ICubeRenderer ` from a `Block`( maybe I should use other than a `Block`). It's a copy of BlockFrame class because I need globally the same thing

Reply:
still that, If you can help me or give me a trace I can follow ...